lib/pipelineTemplateVersionFactory.js

```diff
@@ -130,6 +130,46 @@ class PipelineTemplateVersionFactory extends BaseFactory {
         return newPipelineTemplateVersion;
     }
 
+    /**
+     * List all the versions of a pipeline template
+     * @method create
+     * @param  {Object}     config               Config object
+     * @param templateMetaFactory
+     * @param  {Datastore}  config.datastore     Object that will perform operations on the datastore
+     * @param  {String}     config.name          The template name
+     * @param  {String}     config.namespace     The template namespace
+     * @param  {String}     config.templateId    Identifier of the template
+     * @return {Promise}                         List of PipelineTemplateVersionModel
+     */
+    async list(config, templateMetaFactory) {
+        const { name, namespace } = config;
+        let { templateId } = config;
+
+        const params = {};
+
+        if (templateId) {
+            params.id = templateId;
+        }
+        if (name && namespace) {
+            params.name = name;
+            params.namespace = namespace;
+        }
+
+        const pipelineTemplateMeta = await templateMetaFactory.get(params);
+
+        if (!pipelineTemplateMeta) {
+            return [];
+        }
+
+        templateId = pipelineTemplateMeta.id;
+
+        return super.list({
```

Comment:
We may also have to pass other configs Ex: `config.paginate`

Reply:
ok, so we do intend to support pagination and sorting?